sdk/core/src/main/java/com/klaviyo/core/lifecycle/KlaviyoLifecycleMonitor.kt

```diff
@@ -2,21 +2,26 @@ package com.klaviyo.core.lifecycle
 
 import android.app.Activity
 import android.app.Application
+import android.content.ComponentCallbacks
+import android.content.res.Configuration
 import android.os.Bundle
 import com.klaviyo.core.Registry
+import com.klaviyo.core.utils.WeakReferenceDelegate
 import java.util.Collections
 
 /**
  * Service for monitoring the application lifecycle and network connectivity
  */
-internal object KlaviyoLifecycleMonitor : LifecycleMonitor, Application.ActivityLifecycleCallbacks {
+internal object KlaviyoLifecycleMonitor : LifecycleMonitor, Application.ActivityLifecycleCallbacks, ComponentCallbacks {
 
     private var activeActivities = 0
 
     private val activityObservers = Collections.synchronizedList(
         mutableListOf<ActivityObserver>()
     )
 
+    override var currentActivity: Activity? by WeakReferenceDelegate()
```

Comment:
nit: Is a weak reference even necessary? Seems we're clearing out the activity on pause/stop just fine. I get a weak reference isnt enough on its own cause we cant guarantee garbage collection will run in time but I'm just a little confused on if we need this reference to be any different given the logic below.

Honestly it likely affects nothing so its whatever

Reply:
Agreed, but basically just erring on the side of caution